tests/typ/compiler/ops.typ

```diff
@@ -301,3 +301,41 @@
 #let inc2 = inc.with(y: 2)
 #test(inc2(2), 4)
 #test(inc2(2, y: 4), 6)
+
+---
+// Test chained comparison operations
+
+#test(1 < 2 < 3, true)
+#test(1 < 2 < 1, false)
+#test(1 < 2 <= 2, true)
+#test(1 < 2 <= 1, false)
+#test(1 <= 2 < 3, true)
+#test(1 <= 2 < 1, false)
+#test(1 <= 2 <= 2, true)
+#test(1 <= 2 <= 1, false)
+#test(1 != 2 == 2, true)
+#test(1 != 2 == 1, false)
+#test(1 == 2 != 3, false)
+#test(1 == 2 != 1, false)
+#test(1 == 2 == 2, false)
+#test(1 == 2 == 1, false)
+
+---
+// Test chained comparison with more complex expressions
+#let a = (1, 2, 3)
+#let b = (a: 1, b: 2, c: 3)
+
+#test(a.at(0) < a.at(1) < a.at(2), true)
+#test(a.at(0) < b.a * 2 < a.at(2), true)
+#test(a.at(0) < b.c * 2 < a.at(2), false)
+
+---
+// Test invalid chained operations
+
+// Error: 7-8 only transitive comparisons are allowed
+#test(1 > 2 < 3)
+
+#let (a, b, c, d, e) = (1, 2, 3, 4, 5)
+
+// Error: 7-8 can only chain at most two comparisons
+#test(a < b > c == d != e)
```

Comment:
the error spans seem not ideal. they should cover the whole comparison or at least the faulty operator, not the first operand.